scripts/Inelastic/CrystalField/function.py

```diff
@@ -82,6 +84,21 @@ def attr(self):
     def param(self):
         return self._params
 
+    def fix(self, *args):
+        """
+        Set fixes for the parameters in the function.
+
+        @param args: A list of parameters to fix. Specifying 'all' will fix all of the parameters in a function.
+        """
+        if "all" in [param.lower() for param in args]:
```

Comment:
If a user passes something that isn't a string they will get a vague error about lower not being present.

What about something like:

```python
for arg in args:
    if not isinstance(arg, str):
         logger.warnings(f"Expected a string, found {repr(arg)} instead, ignoring.")
args = [a for a in args if a isinstance(str)]

if "all" in ...
```

Reply:
Also `fix` will not produce a warning if the user forgets to add any params, we want to steer the user to do the right thing rather than silently doing the obviously wrong thing:

```python
if not args:
   logger.warning("No parameters were passed to fix. If you intended to fix all use: fix('all')
   return
```